cloudsync_onedrive.py

```diff
@@ -340,12 +340,27 @@ def _set_drive_list(self):
             log.error("failed to get personal drive info: %s", repr(e))
             raise CloudTokenError("Invalid account, or no onedrive access")
 
+        #shared items
+        shared = self._direct_api("get", "/me/drive/sharedWithMe")
+        for item in shared.get("value", []):
+            try:
+                if not "folder" in item:
+                    # ignore shared files
+                    continue
+                drive_id = item["remoteItem"]["parentReference"]["driveId"]
+                path = urllib.parse.unquote_plus(urllib.parse.urlparse(item["webUrl"]).path).split('/')
+                if len(path) == 5:
```

Comment:
What's the significance of 5?

Reply:
its a magic number... the path looks something like /sites/site-name/drive-name/path/to/folder -- 5 filters out all folders that are not direct children of the root

guess i could make that comment more descriptive